Add Post level components to allow easy building of page level blocks

fabiankaegy opened the pull request:
### High-Level Goal

WordPress itself ships a lot of `Post-` level blocks. These are atomic building blocks that are mostly usefull within the site editor and post templates inside of the Query Loop block.

The issue though is that often times on tightly controlled client builds we want to be able to have more control over the actual markup generated by these elements. And the fact that blocks only support one InnerBlock area means that we cannot achieve every layout and markup with the `Post-` level blocks.

In order to combat this the idea in this PR is to introduce Block Components that resemble the functionality of the `Post-` level component that can be used freely inside any block and take care of all the complex logic required to get these to read/write the correct data from the global post and or query loop post template. 

### The Components:

- `PostTitle`
- `PostFeaturedImage`
- `PostExcerpt`
- `PostAuthor`
    - `PostAuthor.Avatar`
    - `PostAuthor.Name`
    - `PostAuthor.LastName`
    - `PostAuthor.Bio`
    - `PostAuthor.Email`
- `PostTermList`
    - `PostTermList.ListItem`
    - `PostTermList.TermLink`
- `PostCategoryList`
    - `PostCategoryList.ListItem`
    - `PostCategoryList.TermLink`
- `PostPrimaryTerm` _(Requires Yoast SEO for the Primary Term functionality)_
- `PostPrimaryCategory` _(Requires Yoast SEO for the Primary Term functionality)_
- `PostDate`
- `PostContext`
- `PostMeta`

#### Component Structure
These components can be grouped into two categories. There are the components which save and display a singular entity like the `PostTitle`, `PostExcerpt`, `PostPrimaryTerm` etc. These pass through all of the attributes to the actual HTML element and therefore allow full customization of the markup straight out of the box.

```js
<figure className="wp-block-example-hero__media">
    <PostFeaturedImage className="wp-block-example-hero__image" />
</figure>
```

```js
<PostTitle className="wp-block-example-hero__title" tagName="h2" />
```

But then there also are components that can render multiple pieces of information at once. Some examples of this are the `PostAuthor` or `PostTerms`. These can also be used in the same way but then you don't have full control over the markup they produce. In order to still have full control over the markup these components accept render functions or even child components in order to build the markup.

```js
<PostAuthor className="wp-block-example-hero__author">
    <PostAuthor.Avatar className="wp-block-example-hero__author-avatar" />
    <PostAuthor.Name className="wp-block-example-hero__author-name" />
    <PostAuthor.FirstName className="wp-block-example-hero__author-first-name" />
    <PostAuthor.LastName className="wp-block-example-hero__author-last-name" />
    <PostAuthor.Bio className="wp-block-example-hero__author-bio" />
</PostAuthor>
```

```js
<PostCategoryList className="wp-block-example-hero__categories">
    <PostCategoryList.ListItem className="wp-block-example-hero__category">
        <PostCategoryList.TermLink className="wp-block-example-hero__category-link" />
    </PostCategoryList.ListItem >
</PostCategoryList>
```

All these components are reading the current `postId` / `postType` of the post that is getting edited. This however can be changed by wrapping the `Post` components in the `PostContext` wrapper which makes the `postId`, `postType` and whether or not the elements should be editable to all the elements wrapped within. 

![image](https://user-images.githubusercontent.com/20684594/207295119-c0a431a2-2f0b-4aae-bea1-0a076143c87d.png)


### Examples

All the `Post` components are reading/writing data to a post instead of an attribute. They can be used in any block and can inherit both the global post (currently being edited) or also the current post when the block these components are getting used in is a descendant of a query loop.

If you just want to use the global post you can use these components like shown in the `Hero` block in the `examples`

```js
<header {...blockProps}>
    <figure className="wp-block-example-hero__media">
        <PostFeaturedImage className="wp-block-example-hero__image" />
    </figure>
    <PostPrimaryCategory className="wp-block-example-hero__category" />
    <PostTitle className="wp-block-example-hero__title" />
    <PostCategoryList className="wp-block-example-hero__categories">
        <PostCategoryList.ListItem className="wp-block-example-hero__category">
            <PostCategoryList.TermLink className="wp-block-example-hero__category-link" />
        </PostCategoryList.ListItem >
    </PostCategoryList>
    <PostDate className="wp-block-example-hero__date" />
    <PostExcerpt className="wp-block-example-hero__excerpt" />
    <PostAuthor className="wp-block-example-hero__author">
        <PostAuthor.Avatar className="wp-block-example-hero__author-avatar" />
        <PostAuthor.Name className="wp-block-example-hero__author-name" />
        <PostAuthor.FirstName className="wp-block-example-hero__author-first-name" />
        <PostAuthor.LastName className="wp-block-example-hero__author-last-name" />
        <PostAuthor.Bio className="wp-block-example-hero__author-bio" />
    </PostAuthor>
</header>
```

If on the other hand, you want to get the info from the current post within a query loop/post template you can pass in the block context.

```json
{
    "usesContext" : [ "postId", "postType", "queryId" ]
}
```

These values can then be used in the `<PostContext>` component to make the values available to all `Post-` level components used within. 

```js
<article {...blockProps}>
    <PostContext postId={context.postId} postType={context.postType} isEditable={Number.isFinite(context.queryId)}>
        <figure className="wp-block-example-content-item__media">
            <PostFeaturedImage className="wp-block-example-content-item__image" />
        </figure>
        <PostPrimaryCategory className="wp-block-example-content-item__category" />
        <PostTitle className="wp-block-example-content-item__title" />
        <PostCategoryList className="wp-block-example-hero__categories">
            <PostCategoryList.ListItem className="wp-block-example-hero__category">
                <PostCategoryList.TermLink className="wp-block-example-hero__category-link" />
            </PostCategoryList.ListItem >
        </PostCategoryList>
        <PostDate className="wp-block-example-content-item__date" />
        <PostExcerpt className="wp-block-example-content-item__excerpt" />
        <PostAuthor className="wp-block-example-content-item__author">
            <PostAuthor.Avatar className="wp-block-example-content-item__author-avatar" />
            <PostAuthor.Name as="a" className="wp-block-example-content-item__author-name" />
            <PostAuthor.FirstName className="wp-block-example-content-item__author-first-name" />
            <PostAuthor.LastName className="wp-block-example-content-item__author-last-name" />
            <PostAuthor.Bio className="wp-block-example-content-item__author-bio" />
            <PostAuthor.Email className="wp-block-example-content-item__author-email" />
        </PostAuthor>
    </PostContext>
</article>
```

Closing #177 


### Testing Instructions

If you want to test this PR you can either clone this repo and build something in the `example` project bundled with the repo. Or you can install this prerelease version like so:

```bash
npm install @10up/block-components@1.14.6-alpha.4
```

Props: @fabiankaegy @Antonio-Laguna

fabiankaegy committed:
add new property to image component to show placeholder without upload controls

fabiankaegy committed:
add use post hook to get correct post within query loop

fabiankaegy committed:
add useCanEditEntity hook

fabiankaegy committed:
add useIsSupportedTaxonomy hook

fabiankaegy committed:
add hooks to get selected terms

fabiankaegy committed:
add useIsPluginActive hook to check whether a plugin is active

fabiankaegy committed:
add usePrimaryTerm hook which works with Yoast SEO Primary Term Feature

fabiankaegy committed:
add usePropover hook to easily add popovers to elements

fabiankaegy committed:
add PostTitle component

fabiankaegy committed:
add PostFeaturedImage component

fabiankaegy committed:
add PostExcerpt component

fabiankaegy committed:
add PostAuthor component

fabiankaegy committed:
add PostDate and PostDatePicker components

fabiankaegy committed:
add PostCategories component

fabiankaegy committed:
add PostPrimaryCategory component which interacts with Yoast SEO primary term

fabiankaegy committed:
add PostTerms component

fabiankaegy committed:
refactor PostCategories component to use PostTerms

fabiankaegy committed:
add PostPrimaryTerm component

fabiankaegy committed:
refactor PostPrimaryCategory with PostPrimaryTerm

fabiankaegy committed:
add example block for post title component

fabiankaegy committed:
add example post for post featured image component

fabiankaegy committed:
add exampel block for a content item / card component using the post components in a query loop

fabiankaegy committed:
add an example hero block using the post level components

fabiankaegy committed:
install Yoast SEO plugin in the example project

fabiankaegy committed:
fix popover linting issue

fabiankaegy committed:
1.14.2-alpha.0

fabiankaegy committed:
fix use tagName as prop name over as

fabiankaegy committed:
refactor post term list

fabiankaegy committed:
fix render callback and default markup of term list

fabiankaegy committed:
logical refactors

fabiankaegy committed:
add base readme files

fabiankaegy committed:
fix title rendering performance

fabiankaegy committed:
1.14.2-alpha.1

fabiankaegy committed:
add PostContext component

fabiankaegy committed:
fix propTypes of PostCategoryList

fabiankaegy committed:
fix propTypes of PostPrimaryCategory

fabiankaegy committed:
fix refactor components away from passign through context

fabiankaegy committed:
fix import path for all components

fabiankaegy committed:
fix use correct isEditable factor

fabiankaegy committed:
refactor post context context

fabiankaegy committed:
fix remove unused proptypes

fabiankaegy committed:
fix use inert for disabled elements

fabiankaegy committed:
allow usage of custom date format

fabiankaegy commented:
**Status:**

The initial setup of these `Post` level components is ready for review. They aim to provide a simple declarative and composable toolset for anyone trying to write "Hero" or "Card" blocks that can be used anywhere within custom block development. 

The goal here is for a lot of these pieces to get used by the blocks in [UI Kit](https://github.com/10up/ui-kit). But because 90% of the functionality is then wrapped up in these components that means it is easier for anyone that needs to build their own because UI Kit doesn't cover all design quirks to build their own variation of any of these blocks super easily. And with the same editorial experience and robust code powering them.

fabiankaegy commented:
One of the things that I would love to solve at some point but that isn't a blocker for this PR is having PHP helpers which bring the same functionality as the Block Components. Currently you still have to manually get all the correct data in PHP in the block markup.

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	package-lock.json
#	package.json

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	package-lock.json
#	package.json

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	components/index.js
#	hooks/index.js

fabiankaegy committed:
fix remove duplicate import caused by false merge resoluton

fabiankaegy committed:
ensure yoast SEO panel is hidden before running tests

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	components/image/readme.md
#	package-lock.json
#	package.json

fabiankaegy committed:
1.14.4-alpha.0

fabiankaegy committed:
1.14.4-alpha.1

fabiankaegy committed:
1.14.4-alpha.2

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	package-lock.json
#	package.json

fabiankaegy committed:
1.14.5-alpha.0

fabiankaegy committed:
fix always register icons only when the browser is ready

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

fabiankaegy committed:
removed unused hook

fabiankaegy committed:
add readme for useAllTerms hook

fabiankaegy committed:
add readme for useIsPluginActive hook

fabiankaegy committed:
add readme for useIsSupportedTaxonomy hook

fabiankaegy committed:
add readme for usePopover hook

fabiankaegy committed:
add readme to usePost hook

fabiankaegy committed:
add readme for usePrimaryTerm hook

fabiankaegy committed:
fix description of usePrimaryTerm readme

fabiankaegy committed:
add readme for useSelectedTermIds hook

fabiankaegy committed:
add readme for useSelectedTerms hook

fabiankaegy committed:
add readme for useSelectedTermsOfSavedPost hook

fabiankaegy committed:
fix Author components markup

fabiankaegy committed:
add more docs to PostAuthor component

fabiankaegy committed:
add more detailed docs

fabiankaegy committed:
add new hooks and components to main readme

fabiankaegy committed:
fix formatting in readme

fabiankaegy committed:
add more detail to PostCategoryList readme

fabiankaegy committed:
add more context

fabiankaegy committed:
Add yoast warning

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	package-lock.json
#	package.json

fabiankaegy commented:
@Antonio-Laguna @tlovett1 @joesnellpdx This is ready for initial review. Ideally, I would love to get this first iteration merged in soonish to be able to test it in some more projects and then continue building out some of these more advanced use cases.

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

# Conflicts:
#	package-lock.json
#	package.json

fabiankaegy committed:
1.14.6-alpha.0

fabiankaegy committed:
add docs for subcomponents of post term list

fabiankaegy committed:
expand documentation of PostContext component

fabiankaegy committed:
add more context to post context docs

fabiankaegy committed:
add more context to post featured image component

fabiankaegy committed:
fix issue in title component readme

fabiankaegy committed:
add PostMeta component

fabiankaegy committed:
add example block for PostMeta component

fabiankaegy committed:
fix allow usage of render function inside post meta component

fabiankaegy committed:
fix docs of postMeta component

fabiankaegy commented:
@Antonio-Laguna @tlovett1 @joesnellpdx @nicholasio This is ready for review. I have added all the documentation and tested the components 👍

fabiankaegy committed:
fix link component test

fabiankaegy commented:
@10up/frontend-leads Curious whether any of you have any thoughts on the new components I'm adding with this PR :)

fabiankaegy committed:
1.14.6-alpha.1

fabiankaegy committed:
Merge branch 'develop' into feature/post-level-components

fabiankaegy committed:
fix refacor useIsPluginActive to only query for individual plugin

fabiankaegy committed:
1.14.6-alpha.2

fabiankaegy committed:
fix support network-active plugins in useIsPluginActive hook

fabiankaegy committed:
1.14.6-alpha.3

fabiankaegy committed:
add new useTaxonomy hook

fabiankaegy committed:
fix use correct hierarchical or flat term selector

fabiankaegy committed:
fix import path in example

fabiankaegy committed:
1.14.6-alpha.4

fabiankaegy committed:
fix add missing reference to main readme file

Antonio-Laguna commented on components/post-context/readme.md:
```diff
@@ -0,0 +1,51 @@
+# `PostContext`
+
+The `PostContext` component allows you to customize the post object referenced by and of the components referencing the current post object. They are all prefixed with `Post`.
```
```suggestion
The `PostContext` component allows you to customize the post object referenced by any of the components referencing the current post object. They are all prefixed with `Post`.
```

Antonio-Laguna commented on components/post-context/readme.md:
````diff
@@ -0,0 +1,51 @@
+}
+```
+
+*Note: If you enable `isEditable` prop in the `<PostContext>` component for posts that aren't the current post this will actually update the external post. This should be used with caution since it may not be immediately obvious for users.*
````
This is great and I love the warning, I wonder if it makes sense to you to have an `onChange` handler for these. I remember I did something similar to this effort (but less organised and only title and description) where the intention was to let people override the used title/description for editorial purposes but not actually the real title one. `onChange` would offer an escape hatch so you can do as you wish.

Feel free to discard this comment if you don't think it makes sense

fabiankaegy replied:
I updated the readme description here a bit ([b715084](https://github.com/10up/block-components/pull/178/commits/b715084e2069af120d7cf8b8316f43ceb0439060)). There has been quite some improvement in core for how this gets handled. With this new handling in place I don't think this is as much of a problem anymore. 

However I do think the addition of a way to override the change handler to be able to have block locallized overrides is very cool! I will do that in a follow up PR so we can wrap this initial work up  👍 Great idea 🚀

Antonio-Laguna commented on components/post-context/readme.md:
````diff
@@ -0,0 +1,51 @@
+}
+```
+
+The `PostContext` component works great with the Core Query Loop block if you want / need to create a custom block to be used within the post template. Any block that gets used inside the query loop can access the current post id, post type, and query id via the block context. These values can then be used within the `<PostContext>` to make them available for any of the post level components nest ...
````
Besides this example, in the description of the PR you mention `usesContext`, perhaps mention that and either do a quick demo or even link to Gutenberg training material (not sure if we have this covered)

fabiankaegy replied:
I added some more inline context here [fcb479e](https://github.com/10up/block-components/pull/178/commits/fcb479e97de9f9c793490553321ac6b9418e13bf)

Antonio-Laguna commented on components/post-meta/readme.md:
```diff
@@ -0,0 +1,95 @@
+The Component allows you to showcase any piece of post-meta information of the current post.
+
+> **Warning**
+> In order for the Meta Value to show up it needs to be registered properly and have `show_in_rest` set to `true`.
```
🙌 ❤️

Antonio-Laguna commented on components/post-meta/readme.md:
```diff
@@ -0,0 +1,95 @@
+> **Warning**
+> Currently only `string`, `number`, `boolean` for `single` meta values are supported.
+
+You can also completely customize the UI of the meta field by passing a render function to the `PostMeta` component:
```
This is rather known as Function as child which is a pattern derivated from https://reactjs.org/docs/render-props.html I'd clarify this and add some links since this is not react entry level material. Feel free to adjust though!

```suggestion
You can also completely customize the UI of the meta field by passing a [render function as the children](https://reactpatterns.js.org/docs/function-as-child-component/) of the `PostMeta` component:
```

Antonio-Laguna commented on components/post-term-list/readme.md:
```diff
@@ -0,0 +1,71 @@
+
+## Child Components
+
+You can use this component in three different ways. Without any children it will just return some default markup containing the name of the term. Alternatively you can pass a render function as the children. This render function will get the term object passed into it. The third option is using the provided sub-components as shown in the example code. Each of these child component again automatic ...
```
Nice one here!

Antonio-Laguna commented on components/author/index.js:
```diff
@@ -0,0 +1,145 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```
I don't love this. What was the reason to disable these?

fabiankaegy replied:
Ohhh gosh no. That rule should never be disabled. 😱 What was I thinking  😂 Will review 👍

fabiankaegy replied:
Fixed in: [61d03b3](https://github.com/10up/block-components/pull/178/commits/61d03b3774a46114939b1dd7b71e7af2602f0282)

I think I initially had some issues with the linter not catching that these components were components when I had them as `Component.SubComponent = ...`. In order to fix that I refatored it and forgot to remove the disableing of the rule.

Antonio-Laguna commented on components/author/index.js:
```diff
@@ -0,0 +1,145 @@
+};
+
+export const FirstName = (props) => {
+	const { tagName: TagName = 'span', ...rest } = props;
```
[nit] Here and below. 

Don't need a default here since it's added below on `defaultProps`

fabiankaegy replied:
Fixed in [35d0436](https://github.com/10up/block-components/pull/178/commits/35d043649878155c360b3a65f92d6271f2c3174e)

Antonio-Laguna commented on components/author/context.js:
```diff
@@ -0,0 +1,3 @@
+import { createContext } from '@wordpress/element';
+
+export const AUTHOR_CONTEXT = createContext();
```
This is a bit nit but I feel React Contexts (and any React element) should follow PascalCase nomenclature which is standard on React. It feels odd when it's rendering on other elements

Antonio-Laguna replied:
I see this pattern being repeated over the PR, if you think it's actionable then it'd be good to normalise!

fabiankaegy replied:
Updated to match the pattern

See:
- [474dba2](https://github.com/10up/block-components/pull/178/commits/474dba26bde6a23aa9d0eab3bea087be76784c6a)
- [437b832](https://github.com/10up/block-components/pull/178/commits/437b832dc4b139b89600dbe08284aa42018e7cff)
- [31e668a](https://github.com/10up/block-components/pull/178/commits/31e668ac27962549d2f68208df1cf738015fbd85)

fabiankaegy committed:
fix remove disabeling of eslint rule

Antonio-Laguna commented on components/post-date/index.js:
```diff
@@ -0,0 +1,82 @@
+	const settings = getSettings();
+	// To know if the current time format is a 12 hour time, look for "a".
+	// Also make sure this "a" is not escaped by a "/".
+	const is12Hour = /a(?!\\)/i.test(
```
😂 nice trick!

Antonio-Laguna commented on components/post-date/index.js:
```diff
@@ -0,0 +1,82 @@
+};
+
+export const PostDate = (props) => {
+	const { placeholder = __('No date set', 'tenup'), format = undefined, ...rest } = props;
```
Same here with the defaults, in the event of this being translated it would show up two times too

fabiankaegy replied:
Fixed in [c853dff](https://github.com/10up/block-components/pull/178/commits/c853dff4a098d3465b3bd8f0aa3a333ffd29ce64)

Antonio-Laguna commented on components/post-primary-category/index.js:
```diff
@@ -0,0 +1,21 @@
+
+export const PostPrimaryCategory = PostPrimaryTerm;
+
+PostPrimaryCategory.defaultProps = {
+	placeholder: __('Select a category', 'tenup'),
+};
```
These seem duplicated with L17:L21

fabiankaegy replied:
Fixed in [e278bce](https://github.com/10up/block-components/pull/178/commits/e278bce9b9aabea97b5cf056e4dc4e06e09f3ced)

Antonio-Laguna commented on components/post-primary-term/index.js:
```diff
@@ -0,0 +1,47 @@
+
+export const PostPrimaryTerm = (props) => {
+	const {
+		taxonomyName = 'category',
+		placeholder = __('Select a term', 'tenup'),
+		isLink = true,
```
Might add `category` as a defaultProps here and remove the defaults here too as suggested elsewhere.

fabiankaegy replied:
fixed in [27c0ffc](https://github.com/10up/block-components/pull/178/commits/27c0ffce7b057bbdc29edb3607fd4eaffefdde41)

Antonio-Laguna commented on example/src/blocks/hero/index.js:
```diff
@@ -0,0 +1,51 @@
+	},
+	transforms: {},
+	variations: [],
+	parent: ['core/post-content'],
```
TIL !

Antonio-Laguna commented on example/src/blocks/post-featured-image/index.js:
```diff
@@ -0,0 +1,31 @@
+    attributes: {},
+    transforms: {},
+    variations: [],
+	usesContext: [ 'postId', 'postType', 'queryId' ],
```
[nit] Seems like there's an indentation diff here

fabiankaegy replied:
Fixed in 9558467

Antonio-Laguna commented on hooks/use-selected-terms-of-saved-post/readme.md:
```diff
@@ -0,0 +1,20 @@
+    const { context } = props;
+    const { postId } = context;
+    const [selectedCategoriesOfSavedPost, hasResolvedSelectedCategoriesOfSavedPost] =
+  useSelectedTermsOfSavedPost('category', postId);
```
[nit] Indentation issue here

fabiankaegy replied:
Fixed in [5d416d9](https://github.com/10up/block-components/pull/178/commits/5d416d99c0d37deb4a01841b58d328d51567f517)

Antonio-Laguna reviewed (commented):
@fabiankaegy awesome work as usual! 

I've left some minor comments and one suggestion for improvement with `onChange`. I think aside from some minor things this is good to go as is and the rest can be follow up too.
I didn't add comments on every repeated occurrence of the `defaultProps` subject but I can do that too if useful.

I can't wait for this to be out!

fabiankaegy committed:
fix rename PostContext to use PascalCase

fabiankaegy committed:
fix rename PostTermItemContext to use PascalCase

fabiankaegy committed:
fix rename AuthorContext to use PascalContext

fabiankaegy committed:
fix delete unused context definition

fabiankaegy committed:
fix rename PostTermItemContext to PostTermContext

fabiankaegy committed:
fix remove redundant default value covered by defaultProps

fabiankaegy committed:
add more detail to PostContext readme about `usesContext`

fabiankaegy committed:
add more context to PostContext readme about updating external post

fabiankaegy committed:
Update components/post-context/readme.md

Co-authored-by: Antonio Laguna <antonio.laguna@10up.com>

fabiankaegy committed:
Update components/post-meta/readme.md

Co-authored-by: Antonio Laguna <antonio.laguna@10up.com>

fabiankaegy committed:
fix indentation of code example in readme

fabiankaegy committed:
fix formatting issues in post featured image block example

fabiankaegy committed:
fix use defaultProps over inline defaults in PostPrimaryTerm component

fabiankaegy committed:
fix remove duplicate defaultProps definition in PostPrimaryCategory component

fabiankaegy committed:
fix rely on defaultProps over inline defaults in PostDate component

fabiankaegy committed:
fix rely on defaultProps over inline default in PostExcerpt component

fabiankaegy committed:
fix rely on defaultProps over inline default in MetaString component

fabiankaegy committed:
fix rely on defaultProps over inline default in PostTermList component

fabiankaegy committed:
fix rely on defaultProps over inline default in ListItem component

fabiankaegy committed:
fix rely on defaultProps over inline default in PostTitle component

fabiankaegy committed:
1.14.6-alpha.5

fabiankaegy commented:
Thank you so much @Antonio-Laguna for you thorough feedback :) I've addressed all your feedback and think I got all the instances of the `defaultProps` definition :) 

As mentioned above I will create a follow up for the `onChange` idea :) 

Once the CI checks are passing I will merge this into develop as the initial round ready for some more wider testing :)
